segmentstore/server/src/main/java/io/pravega/segmentstore/server/containers/StreamSegmentContainer.java

```diff
@@ -593,7 +602,7 @@ public boolean isOffline() {
                                     sourceMetadata.getLength(), sourceMetadata.getAttributes()));
                 } else {
                     // Source now has some data - we must merge the two.
-                    MergeSegmentOperation operation = new MergeSegmentOperation(targetSegmentId, sourceSegmentId);
+                    MergeSegmentOperation operation = new MergeSegmentOperation(targetSegmentId, sourceSegmentId, attributeUpdates);
```

Comment:
This one is a bit tricky. Check how we process `StreamSegmentAppendOperation` and  `UpdateAttributesOperation`.

You need to change this method to invoke `processAttributeUpdaterOperation` as there are some scenarios which will not work out of the box - that method takes care of it for you.